### PR TITLE
Vectorize mem fixes

### PR DESCRIPTION
**Pull request by abialas1**
Added some fixes to vectorize mem I found:
1. Make the optimization checks to be stricter (if any vector load or vector reduce in the loop cannot be vectorized then do not do optimization)
2. Check if the YieldOp type matches the loop result type, if not then update it
3. Change after code review from access qualifiers

### CODE REVIEW

File: pmlc/dialect/pxa/transforms/vectorize_mem.cc [hunk 1]
```diff
@@ -320,6 +325,11 @@ struct VectorizeMemImpl {
     auto results = loopOp.results();
     for (auto res : results) {
       res.setType(nemMemrefType);
+      // Update also AffineYieldOp if type does not match
+      for (auto yieldOp : loopOp.getOps<AffineYieldOp>()) {
+        if (yieldOp.getOperand(0).getType() != nemMemrefType)
```
Comment (jbruestle) [inline pmlc/dialect/pxa/transforms/vectorize_mem.cc:330]: Seems like the original code had a typo (I presume nemMemrefType is supposed to be newMemrefType). Probably should fix the name as part of this commit.

**Review by jbruestle — COMMENTED** [review]
Looks good and makes sense, tiny typo fix request.

File: pmlc/dialect/pxa/transforms/vectorize_mem.cc [hunk 2]
```diff
@@ -295,9 +300,9 @@ struct VectorizeMemImpl {
         VectorType::get({memOpsPlan.subgroupSize * tileSize},
                         vectorReduce.getVectorType().getElementType());
 
-    auto nemMemrefType =
+    auto memMemrefType =
```
Comment (flaub) [inline pmlc/dialect/pxa/transforms/vectorize_mem.cc:303]: This should probably be renamed to `newMemrefType`

Comment (abialas1) [reply pmlc/dialect/pxa/transforms/vectorize_mem.cc:303]: you are right! let me change it

**Review by flaub — APPROVED** [review]

**Pull request merged**